File: app/src/main/java/com/tutv/android/ui/tv_poster_list/tv_poster/TvPosterPresenter.java
package com.tutv.android.ui.tv_poster_list.tv_poster;

import com.tutv.android.domain.Series;
import com.tutv.android.repository.SeriesRepository;

import java.lang.ref.WeakReference;

public class TvPosterPresenter {
    private final WeakReference<TvPosterView> view;
    private final SeriesRepository seriesRepository;

    private Series series;

    public TvPosterPresenter(TvPosterView view, SeriesRepository seriesRepository) {
        this.view = new WeakReference<>(view);
        this.seriesRepository = seriesRepository;
    }

    public void bind(Series series) {
        this.series = series;

        if(view.get() != null) {
            view.get().setImageUrl(series.getPosterUrl());
            view.get().setText(series.getName());
        }
    }

    public void onClick() {
        if(view.get() != null)
            view.get().navigateToSeriesPage(series.getId());
    }
}

File: app/src/main/java/com/tutv/android/ui/tv_poster_list/tv_poster/TvPosterPresenter.kt
package com.tutv.android.ui.tv_poster_list.tv_poster

import com.tutv.android.domain.Series
import com.tutv.android.repository.SeriesRepository
import java.lang.ref.WeakReference

class TvPosterPresenter(
        view: TvPosterView,
        var seriesRepository: SeriesRepository) {

    private val view: WeakReference<TvPosterView?> = WeakReference(view)
    private var series: Series? = null

    fun bind(series: Series?) {
        this.series = series
        view.get()?.setImageUrl(series?.posterUrl)
        view.get()?.setText(series?.name)
    }

    fun onClick() {
        view.get()?.navigateToSeriesPage(series?.id)
    }

}